var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -52,8 +56,8 @@ class Cuda(Package):
     # Mojave support -- only macOS High Sierra 10.13 is supported.
     conflicts('arch=darwin-mojave-x86_64')
 
-    def setup_environment(self, spack_env, run_env):
-        run_env.set('CUDA_HOME', self.prefix)
+    def setup_run_environment(self, env):
+        env.set('CUDA_HOME', self.prefix)
```

Comment:
This looks like the reason for #13969 to me.

Reply:
I don't think that is the reason since the same environment was set before.
The underlying reason probably stems from the introduction of `libcu++` with CUDA 10.2, which ships its own device-enabled stdlib.

Reply:
Could be the `find_libraries('libcuda', ..., recursive=True)` thing below that was introduced by @scheibelp in #11449